### PR TITLE
Correct rand floating point documentation

### PR DESCRIPTION
**Pull request by Lperlind**
This corrects:
* `(n, m]` to be the correct notation of `[n, m)`
* Comments the lack of guarantee with floating point with regards to being inclusive/exclusive.

### CODE REVIEW

**Comment by Lperlind** [comment]
Looks like I am on a stale version of odin

File: core/math/rand/rand.odin [hunk 1]
```diff
@@ -525,13 +525,14 @@ int_max :: proc(n: int, r: ^Rand = nil) -> (val: int) {
 }
 
 /*
-Generates a random double floating point value in the range `(0, 1]` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Generates a random double floating point value in the range `0 to 1` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Due to floating point precision there is no guaruntee if the upper and lower bounds are inclusive/exclusive with the exact floating point value.
```
Comment (Kelimion) [inline core/math/rand/rand.odin:529]: Typo: guarunee -> guarantee.

File: core/math/rand/rand.odin [hunk 2]
```diff
@@ -554,13 +555,14 @@ Possible Output:
 @(require_results) float64 :: proc(r: ^Rand = nil) -> (val: f64) { return f64(int63_max(1<<53, r)) / (1 << 53) }
 
 /*
-Generates a random single floating point value in the range `(0, 1]` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Generates a random single floating point value in the range `0 to 1` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Due to floating point precision there is no guaruntee if the upper and lower bounds are inclusive/exclusive with the exact floating point value.
```
Comment (Kelimion) [inline core/math/rand/rand.odin:485]: Same typo.

File: core/math/rand/rand.odin [hunk 3]
```diff
@@ -583,15 +585,16 @@ Possible Output:
 @(require_results) float32 :: proc(r: ^Rand = nil) -> (val: f32) { return f32(float64(r)) }
 
 /*
-Generates a random double floating point value in the range `(low, high]` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Generates a random double floating point value in the range `low to high` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Due to floating point precision there is no guaruntee if the upper and lower bounds are inclusive/exclusive with the exact floating point value.
```
Comment (Kelimion) [inline core/math/rand/rand.odin:511]: Same.

File: core/math/rand/rand.odin [hunk 4]
```diff
@@ -613,15 +616,16 @@ Possible Output:
 */
 @(require_results) float64_range :: proc(low, high: f64, r: ^Rand = nil) -> (val: f64) { return (high-low)*float64(r) + low }
 /*
-Generates a random single floating point value in the range `(low, high]` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Generates a random single floating point value in the range `low to high` using the provided random number generator. If no generator is provided the global random number generator will be used.  
+Due to floating point precision there is no guaruntee if the upper and lower bounds are inclusive/exclusive with the exact floating point value.
```
Comment (Kelimion) [inline core/math/rand/rand.odin:539]: Almost as if it's a copy-paste typo.

**Review by Kelimion — CHANGES_REQUESTED** [review]

File: core/math/rand/rand.odin (hunk 4 above)
Comment (Lperlind) [reply core/math/rand/rand.odin]: It was a copy paste :)

**Review by Kelimion — APPROVED** [review]

**Pull request closed without merging**